imap_processing/hit/l0/decom_hit.py

```diff
@@ -0,0 +1,464 @@
+    #       when the cdf yaml is used to update all the dimension names?
+
+    # Add new data variables to the dataset
+    epoch_science_frame = np.array(epoch_science_frame)
+    # Replace epoch per packet dimension with epoch per science frame dimension
+    sci_dataset = sci_dataset.drop_vars("epoch")
+    sci_dataset.coords["epoch"] = epoch_science_frame
+    sci_dataset["count_rates_binary"] = xr.DataArray(
+        count_rates_binary, dims=["epoch"], name="count_rates_binary"
+    )
+    sci_dataset["pha_binary"] = xr.DataArray(
+        pha_binary, dims=["epoch"], name="pha_binary"
+    )
+    return sci_dataset
+
+
+def decom_hit(sci_dataset: xr.Dataset) -> xr.Dataset:
+    """
+    Group and decode HIT science data packets.
+
+    This function updates the science dataset with
+    organized, decommutated, and decompressed data.
+
+    The dataset that is passed in contains the unpacked
+    CCSDS header and the science data as bytes as follows:
+
+    <xarray.Dataset>
+    Dimensions:       epoch
+    Coordinates:
+      * epoch         (epoch) int64
+    Data variables:
+        sc_tick       (epoch) uint32
+        science_data  (epoch) <U2096
+        version       (epoch) uint8
+        type          (epoch) uint8
+        sec_hdr_flg   (epoch) uint8
+        pkt_apid      (epoch) uint16
+        seq_flgs      (epoch) uint8
+        src_seq_ctr   (epoch) uint16
+        pkt_len       (epoch) uint16
+
+    The science data for a science frame (i.e. 1 minute of data)
+    is spread across 20 packets. This function groups the
+    data into science frames and decommutates and decompresses
+    binary into integers.
+
+    Parameters
+    ----------
+    sci_dataset : xr.Dataset
+        Xarray dataset containing HIT science packets
+        from a CCSDS file.
+
+    Returns
+    -------
+    sci_dataset : xr.Dataset
+        Updated xarray dataset with new fields for all count
+        rates and pulse height event data per valid science frame
+        needed for creating an L1A product.
+    """
+    # Update ccsds header fields to use sc_tick as dimension
+    sci_dataset = update_ccsds_header_data(sci_dataset)
+    # Group science packets into groups of 20
+    sci_dataset = assemble_science_frames(sci_dataset)
+    # Parse count rates data from binary and add to dataset
+    parse_count_rates(sci_dataset)
+
+    # TODO:
+    #  Parse binary PHA data and add to dataset (function call)
+
+    return sci_dataset
+
+
+# TODO: remove main after code is finalized
+if __name__ == "__main__":
+    packet_definition = (
+        imap_module_directory / "hit/packet_definitions/hit_packet_definitions.xml"
+    )
+
+    # L0 file path
+    packet_file = Path(imap_module_directory / "tests/hit/test_data/sci_sample.ccsds")
+
+    datasets_by_apid = packet_file_to_datasets(
+        packet_file=packet_file,
+        xtce_packet_definition=packet_definition,
+    )
+
+    science_dataset = datasets_by_apid[1252]
```

Comment:
You can create ENUM for apid like I have done for other instruments.

Reply:
Good suggestion! I currently have this in hit_l1a.py, which will call this file to perform the decom work on the science packets. I might move the ENUM for apid to this file instead, but haven't decided yet